src/jit/codegenxarch.cpp

```diff
@@ -2651,9 +2638,14 @@ CodeGen::genStoreRegisterReturnInLclVar(GenTreePtr treeNode)
         getStructReturnRegisters(type0, type1, &retReg0, &retReg1);
 
         assert(retReg0 != REG_NA && retReg1 != REG_NA);
-
+        
```

Comment:
Here too we are making an assumption that the return values of the call would be in the right registers, which is not necessarily true.  Please add a comment here similar to the one we added in genStructReturn()